src/PKSim.Infrastructure/ORM/Repositories/DimensionRepository.cs

```diff
@@ -102,18 +102,39 @@ public IDimension MergedDimensionFor(object objectThatMightHaveDimension)
 
       protected override void DoStart()
       {
+         loadDimensionsFromFile();
+
+         addInputDoseDimension();
+
+         _dimensionNames = _dimensionFactory.GetDimensionNames().ToList();
+         _dimensionFactory.AddDimension(Constants.Dimension.NO_DIMENSION);
+      }
+
+      private void addInputDoseDimension()
+      {
+         var inputDoseDimension = _dimensionFactory.AddDimension(new BaseDimensionRepresentation(), CoreConstants.Dimension.InputDose, CoreConstants.Units.KgPerKg);
+         inputDoseDimension.BaseUnit.Visible = false;
+
+         //factor to convert mg to kg
+         addInputDoseUnit(inputDoseDimension, CoreConstants.Units.mg, 1e-6);
+
+         //factor to convert mg/kg to kg/kg
+         inputDoseDimension.DefaultUnit = addInputDoseUnit(inputDoseDimension, CoreConstants.Units.MgPerKg, 1e-6);
+
+         //factor to convert mg/m2 to kg/dm2
+         addInputDoseUnit(inputDoseDimension, CoreConstants.Units.MgPerM2, 1e-8);
```

Comment:
This is the conversion factor to convert mg/m2 (input unit) to kg/dm2 (core unit)